p2p/router.go

```diff
@@ -259,26 +252,31 @@ func (r *Router) acceptPeers(transport Transport) {
 			continue
 		}
 
-		peerID := conn.NodeInfo().NodeID
-		if r.store.Claim(peerID) == nil {
-			r.logger.Error("already connected to peer, rejecting connection", "peer", peerID)
-			_ = conn.Close()
-			continue
-		}
+		go func() {
+			defer func() {
+				_ = conn.Close()
+			}()
 
-		queue := newFIFOQueue()
-		r.peerMtx.Lock()
-		r.peerQueues[peerID] = queue
-		r.peerMtx.Unlock()
+			peerID := conn.NodeInfo().NodeID
+			if err := r.peerManager.Accepted(peerID); err != nil {
+				r.logger.Error("failed to accept connection", "peer", peerID, "err", err)
+				return
+			}
+
+			queue := newFIFOQueue()
+			r.peerMtx.Lock()
+			r.peerQueues[peerID] = queue
+			r.peerMtx.Unlock()
+			r.peerManager.Ready(peerID)
 
-		go func() {
 			defer func() {
```

Comment:
Why not move the `_ = conn.Close()` to this `defer`?

Reply:
Then it wouldn't close the connection if the `peerManager.Accepted()` call above errors (or any other calls we may add later).

Reply:
You're right. I missed that error check.